src/main/java/com/yahoo/bullet/common/Validator.java

```diff
@@ -751,6 +761,29 @@ public static boolean isClassName(Object value) {
         };
     }
 
+    /**
+     * Checks to see if the value is a {@link Map} of the given key and value type.
+     *
+     * @param keyType The class of the key of the map to check for.
+     * @param valueType The class of the value of the map to check for.
+     * @param <K> The type of the key in the map.
+     * @param <V> The type of the value in the map.
+     * @return A Predicate that checks tor see if the value was a non-empty map of the given types.
+     */
+    @SuppressWarnings("unchecked")
+    public static <K, V> Predicate<Object> isMapOfType(Class<K> keyType, Class<V> valueType) {
+        return value -> {
+            if (!isNonEmptyMap(value)) {
```

Comment:
Why do we insist the map is not empty?

Reply:
Generics type erasure. Need key, values to check their types